### ISSUE TITLE
[Feature Req] Check for executable bit being set on scripts

### ISSUE THREAD

**Issue by adrianmace** [koalaman/shellcheck#2127]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/your/interpreter
your script here

```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

Script is not executable: Run chmod +x filename.sh

**Comment by agentgoblin** [comment]
This is good for first glance, but what if the script must have rights -rwxrwxr-- or -rwxrw-rw-? I.e. should only run from owner or group?

Second, the script may be part of the larger scripting solution and used from the main script via "source filename.sh". In this case the script doesn't not need an executable bit.

**Comment by adrianmace** [comment]
> but what if the script must have rights -rwxrwxr-- or -rwxrw-rw-? I.e. should only run from owner or group?

Change the warning to `Script is not executable by world, group, or owner: In most cases this can be fixed by running chmod +x filename.sh`

> the script may be part of the larger scripting solution and used from the main script via "source filename.sh". In this case the script doesn't need an executable bit.

I would hope that this individual check could be disabled just like any other in some way.
